tests/test_browser.py

```diff
@@ -2770,7 +2770,13 @@ def in_html(expected, args=[]):
   def test_glfw3(self):
     for opts in [[], ['-s', 'LEGACY_GL_EMULATION=1'], ['-Os', '--closure', '1']]:
       print(opts)
-      self.btest(path_from_root('tests', 'glfw3.c'), args=['-s', 'USE_GLFW=3', '-lglfw', '-lGL'] + opts, expected='1')
+      self.btest(path_from_root('tests', 'glfw3.c'), args=['-DCLIENT_API=GLFW_OPENGL_ES_API', '-s', 'USE_GLFW=3', '-lglfw', '-lGL'] + opts, expected='1')
+
+  @requires_graphics_hardware
+  def test_glfw3_noapi(self):
+    for opts in [[], ['-s', 'LEGACY_GL_EMULATION=1'], ['-Os', '--closure', '1']]:
+      print(opts)
+      self.btest(path_from_root('tests', 'glfw3.c'), args=['-DCLIENT_API=GLFW_NO_API', '-s', 'USE_GLFW=3', '-lglfw', '-lGL'] + opts, expected='1')
```

Comment:
the code duplication can be avoided by using the `@parameterized` macro, see for example `test_gles2_emulation` in this file.

Reply:
Thanks, I'll check it out.